src/sponge/truncated.rs

```diff
@@ -0,0 +1,158 @@
+        "ce8bcf8952c3daf89ee9fe55ff3acf3bf83c17d28c50fb7fa0db3ce471cc1134",
+        "3ee00d2d773237f5f807715894f1a320019c34914b880d4c87299f83de7ece2e",
+        "3a1eef3d0a84798020b3016ae323f0c71916074b636c6ca55e53abd859dbd10e",
+        "6c4e854816920cc4b34820d6e5d5c4c210125a35289261c42c20beea88375439",
+        "8264f7a36717ab6149bd0c7b2a6496e9aa4952fa74f9e20075d712f61e6c3e12",
+        "0601f84b745cb0ee65ed275a3913566ca2948e8c7911c4c2f2e34ecaa446f23c",
+        "86126b269583662d1ea7c1a9045784dab704c8305218c621483a48aefbd1611c",
+        "56d655c6ae6136b9d7b22824999a182acdf68a8a5a5095e586a5c9038b635511",
+        "3ff4311953234ce812ef86ec4c0f3bf381a4a9d31a9025813ba69e7e3c19021b",
+        "8d9aec8c1b34e5f59ad4633a670e7bede86ef777395c7b14057f28c2c2ae4802",
+        "4f47cd90d7f732b7255dceb56084d0889824b66b929bf57255db3e95786f813a",
+        "535ac1999b63f38bf718ef12b98dd0f095975244aefc402ac6203878d8f6e93c",
+        "e1eb9d629f14b587e6c5eed82aefea704f2968edbb0bedbd906bfa31089f7412",
+        "958318907edb1b919a62fd62aeab05e2c6fea95fc731ba169ae8e406aec5361a",
+        "e111a0664ac113b960cd336643db4b34c5cd4f69de84d44be95cadaca4d19115",
+    ];
+
+    fn truncated_gadget(
+        composer: &mut StandardComposer,
+        messages: &[BlsScalar],
+    ) {
+        let truncated_res = hash(messages);
+        let final_point = JubJubAffine::from(
+            dusk_jubjub::GENERATOR_EXTENDED * &truncated_res,
+        );
+
+        // Circuit
+        let input_vars: Vec<Variable> = messages
+            .iter()
+            .map(|message| composer.add_input(*message))
+            .collect();
+        let truncated_res_var = gadget(composer, &input_vars);
+        let point_result = composer.fixed_base_scalar_mul(
+            truncated_res_var,
+            dusk_jubjub::GENERATOR_EXTENDED,
+        );
+        composer.assert_equal_public_point(point_result, final_point);
+    }
+
+    #[test]
+    fn truncation_sponges_match() -> Result<(), Error> {
+        // Setup OG params.
+        let public_parameters = PublicParameters::setup(1 << 17, &mut OsRng)?;
+        let (ck, vk) = public_parameters.trim(1 << 17)?;
+
+        let test_inputs: Vec<BlsScalar> = TEST_INPUTS
+            .iter()
+            .map(|input| BlsScalar::from_hex_str(input).unwrap())
+            .collect();
+
+        let gadget_tester = |messages: &[BlsScalar]| -> Result<(), Error> {
+            let mut prover = Prover::new(b"sponge_tester");
+            truncated_gadget(prover.mut_cs(), messages);
+            let pi = prover.mut_cs().construct_dense_pi_vec().clone();
+            prover.preprocess(&ck)?;
+            let proof = prover.prove(&ck)?;
+
+            // Verify
+            let mut verifier = Verifier::new(b"sponge_tester");
+            truncated_gadget(verifier.mut_cs(), messages);
+            verifier.preprocess(&ck)?;
+            verifier.verify(&proof, &vk, &pi)
+        };
+
+        assert!(gadget_tester(&test_inputs[0..3]).is_ok());
```

Comment:
Small nit: ranges starting with zeroes are redundant and should be described as `[..3]` instead